src/components/CommentList.jsx

```diff
@@ -16,17 +19,23 @@ export const CommentList = ({comments,comment_count}) =>{
             
                 {viewComments
                 ?<ul id='commentContainer'>
-                    {comments.map(comment=>{
+                    {comments.map((comment,index)=>{
                         return (
                             <li id='commentLayout' key={comment.comment_id}>
+                                {comment.author?
+                                <>
                                 <div id='post'>
-                                    <p>{comment.author}</p>
+                                    <p>{comment.author} -</p>
                                     <p>{new Date(comment.created_at).toDateString()}</p>
                                 </div>
                          
                                 <p>{comment.body}</p>
-
+                                <div>
                                 <div id='likes'><img src={like} alt='like icon'/><p>{comment.votes}</p><img src={dislike} alt='dislike icon'/></div>
+                                {username===author || username===comment.author?<DeleteComment commentPosition = {index} setComments ={setComments} commentId = {comment.comment_id}/>:null}
```

Comment:
This could be a lot dry-er if you de-structured the values from comment on line 22!